client/core/bond.go

```diff
@@ -150,337 +150,304 @@ func minBondLifetime(net dex.Network, bondExpiry int64) time.Duration {
-				c.log.Warnf("Unable to post the required bond while disconnected or account is locked.")
-				continue
-			}
-			_, err = wallet.refreshUnlock()
-			if err != nil {
-				c.log.Errorf("failed to unlock bond asset wallet %v: %v", unbip(bondAssetID), err)
-				continue
-			}
-			if !wallet.synchronized() {
-				c.log.Warnf("Wallet %v is not yet synchronized with the network. Cannot post new bonds yet.",
-					unbip(bondAssetID))
-				continue // otherwise we might double spend if the wallet keys were used elsewhere
-			}
-
-			// For the max bonded limit, we'll normalize all bonds to the
-			// currently selected bond asset.
-			toPost := mustPost
-			amt := bondAsset.Amt * uint64(mustPost)
-			currentlyBondedAmt := uint64(pendingStrength+liveStrength+expiredStrength) * bondAsset.Amt
-			for maxBondedAmt > 0 && amt+currentlyBondedAmt > maxBondedAmt && toPost > 0 {
-				toPost-- // dumber, but reads easier
-				amt = bondAsset.Amt * uint64(toPost)
-			}
-			if toPost == 0 {
-				c.log.Warnf("Unable to post new bond with equivalent of %s currently bonded (limit of %s)",
-					wallet.amtString(currentlyBondedAmt), wallet.amtString(maxBondedAmt))
-				continue
-			}
-			if toPost < mustPost {
-				c.log.Warnf("Only posting %d bond increments instead of %d because of current bonding limit of %s",
-					toPost, mustPost, wallet.amtString(maxBondedAmt))
-			}
-
-			bondLifetime := minBondLifetime(c.net, bondExpiry)
-			lockTime := time.Now().Add(bondLifetime).Truncate(time.Second)
-			if lockDur := time.Until(lockTime); lockDur > lockTimeLimit {
-				c.log.Errorf("excessive lock time (%v>%v) - not posting!", lockDur, lockTimeLimit)
-			} else {
-				c.log.Tracef("Bond lifetime = %v (lockTime = %v)", bondLifetime, lockTime)
-				_, err = c.makeAndPostBond(dc, true, wallet, amt, lockTime, bondAsset)
-				if err != nil {
-					c.log.Errorf("Unable to post bond: %v", err)
-				} // else it's now in pendingBonds
-			}
-		} // mustPost
-	} // range c.dexConnections
+// recordReserveAdjustments computes required reserves adjustments for any
+// unactuated tierChanges
+func recordReserveAdjustments(dc *dexConnection, state *dexAcctBondState, bondAsset *msgjson.BondAsset, netReserveDeltas map[uint32]int64, actuatedTierChanges map[uint32][]acctUpdate) {
+	if state.tierChange != 0 {
+		reserveDelta := bondOverlap * -state.tierChange * int64(bondAsset.Amt)
+		netReserveDeltas[state.bondAssetID] += reserveDelta // negative means free reserves
+		// After applying the net reserveDelta, we would normally *zero* the
+		// tierChange field, but it could change, so we remember it.
+		actuatedTierChanges[state.bondAssetID] = append(actuatedTierChanges[state.bondAssetID],
+			acctUpdate{dc.acct, state.tierChange, reserveDelta},
+		)
+	}
+}
 
+// applyReserveAdjustments notifies wallets of the aggregate reserve
+// adjustments over all dex connections.
+func (c *Core) applyReserveAdjustments(netReserveDeltas map[uint32]int64, actuatedTierChanges map[uint32][]acctUpdate) {
```

Comment:
This body of this reserves helper and `recordReserveAdjustments` look better suited to stay in `rotateBonds`.  Is this also important for ETH bond renewal/extension?

Reply:
I've moved it back there. The point if this was that this function was already ~500 lines and the update bond code will also add a good amount of additional logic. Personally I think it's very hard to wrap my head around these huge functions. If the details are extracted into smaller helper functions, it's easier to quickly know what is happening and make changes. Why do you like the linear flowing code better?

Reply:
> Why do you like the linear flowing code better?

Just readable for me is all.

Regarding the reserves adjustments, when I see variables like `netReserveDeltas` and `actuatedTierChanges` (and these two methods that need them) that are only instantiated or used in one other method, it stands out.  The functionality in either of these methods is useless unless applied exactly how `rotateBonds` applies it, and they'd always be used in succession.  So for me it's confusing (bounce around the code and wonder why they are separate methods), and it adds more the `Core`'s method set, which is not such a good thing even if they are not exported.